### PR TITLE
openssl_pkcs12: allow to specify certificate bundles in other_certificates

### LINKED ISSUE #149: other_certificates is not pulling all certs in file
Here is my ansible cfg

  - name: Generate PFX file
    become: yes
   become_user: "{{ app_user }}"
    community.crypto.openssl_pkcs12:
      action: export
      path: "{{ sslpath }}/{{ fqdn_v }}/{{ fqdn_v }}.pfx"
      friendly_name: "{{ fqdn_v }}"
      privatekey_path: "{{ sslpath }}/{{ fqdn_v }}/{{ fqdn_v }}.key"
      certificate_path: "{{ sslpath }}/{{ fqdn_v }}/{{ fqdn_v }}.crt"
      **other_certificates: "{{ sslpath }}/{{ fqdn_v}}/{{ InCommon_crt }}"**
      passphrase: "{{ pfx_pass }}"
      state: present
      mode: "0644"

the "InCommon_crt" contains the whole chain, but when it goes and reads the .crt it only pulls the first certificate entry it finds:

There are 3 sets of certs in the InCommon.crt file.  

All being and end with:

-----BEGIN CERTIFICATE-----

-----END CERTIFICATE-----

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Does some refactoring of PEM splitting code, and fixes #149.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/module_utils/crypto/identify.py
plugins/module_utils/crypto/pem.py
plugins/modules/acme_certificate.py
plugins/modules/certificate_complete_chain.py
plugins/modules/openssl_pkcs12.py

### CODE REVIEW

**Comment by felixfontein** [comment]
ready_for_review

**Comment by felixfontein** [comment]
CC @Andersson007

File: plugins/modules/openssl_pkcs12.py [hunk 1]
```diff
@@ -27,10 +27,19 @@
         choices: [ export, parse ]
     other_certificates:
         description:
-            - List of other certificates to include. Pre 2.8 this parameter was called C(ca_certificates)
+            - List of other certificates to include. Pre Ansible 2.8 this parameter was called C(ca_certificates)
```
Comment (Andersson007) [inline plugins/modules/openssl_pkcs12.py:30]:
```suggestion
            - List of other certificates to include. Pre Ansible 2.8 this parameter was called I(ca_certificates).
```

File: plugins/modules/openssl_pkcs12.py [hunk 2]
```diff
@@ -27,10 +27,19 @@
...
+              set I(other_certificates_parse_all) to C(true).
         type: list
         elements: path
         aliases: [ ca_certificates ]
+    other_certificates_parse_all:
+        description:
+            - If set to C(true), assumes that the files mentioned in I(other_certificates) can contain more than one
+              certificate per file (or even none per file).
+        type: bool
+        default: false
```
Comment (Andersson007) [inline plugins/modules/openssl_pkcs12.py:41]:
```suggestion
```
;)

**Comment by Andersson007** [comment]
could you also please add an example to the EXAMPLE block

File: plugins/modules/openssl_pkcs12.py (hunk 2 above)
Comment (felixfontein) [reply plugins/modules/openssl_pkcs12.py:41]: The option is defined as `other_certificates_parse_all=dict(type='bool', default=False),` in the argspec, so removing the default here would be wrong, wouldn't it?

Comment (Andersson007) [reply plugins/modules/openssl_pkcs12.py:41]: Oh, i didn't sleep well because of SpaceX;) I mixed it up with `required: no`, sorry:) Irrelevant

**Review by Andersson007 — APPROVED** [review]

**Comment by Andersson007** [comment]
@felixfontein thanks for the feature! LGTM

**Comment by felixfontein** [comment]
@Andersson007 thanks a lot for reviewing this!

**Pull request merged**